### PR TITLE
Enable performance telemetry tracking for production deployments

### PR DESCRIPTION
**Pull request by jwilander**
#### Summary
Per discussion here https://community.mattermost.com/core/pl/seageg6dhidkmmx9rz9urgk9mw.

#### Release Note
```release-note
Enable performance telemetry tracking for production deployments not running in developer mode. Disabled when telemetry is toggled off.
```

### CODE REVIEW

File: actions/telemetry_actions.jsx [hunk 1]
```diff
@@ -14,6 +17,15 @@ const SUPPORTS_MEASURE_METHODS = isSupported([
     performance.clearMeasures,
 ]);
 
+export function isTelemetryEnabled(store = reduxStore) {
```
Comment (jwilander) [inline actions/telemetry_actions.jsx:20]: Not a fan of passing the store in but couldn't find a cleaner method that would allow for unit testing.

Comment (jwilander) [reply actions/telemetry_actions.jsx:20]: Good call, that's much cleaner. I'll update the PR.

**Comment by jwilander** [comment]
@hmhealey PTAL

File: actions/telemetry_actions.jsx [hunk 2]
```diff
@@ -14,6 +17,15 @@ const SUPPORTS_MEASURE_METHODS = isSupported([
     performance.clearMeasures,
 ]);
 
+export function isTelemetryEnabled(state = store.getState()) {
```
Comment (hmhealey) [inline actions/telemetry_actions.jsx:20]: Maybe remove the default value from here? If we're only calling `shouldTrackPerformance` without an argument, I think it'd be best to have that be the only place its done to keep people using the other selectors correctly.

**Review by hmhealey — APPROVED** [review]
One small suggestion, but otherwise looks good

**Review by M-ZubairAhmed — APPROVED** [review]

**Comment by cwarnermm** [comment]
@jwilander - Are [telemetry documentation](https://docs.mattermost.com/manage/telemetry.html) updates needed as part of this PR?

**Comment by jwilander** [comment]
@cwarnermm nope, this falls into the same categories called out in the documentation so all good 👍

**Pull request merged**